### PR TITLE
compliance: Move export screening to RHSSO class (PROJQUAY-2056)

### PR DESCRIPTION
**Pull request by jonathankingfc**
- Created a RHSSO class
- Put export compliance logic in class override

### CODE REVIEW

File: oauth/services/rhsso.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class RHSSOOAuthService(OIDCLoginService):
+    def __init__(self, config, key_name):
+        super(RHSSOOAuthService, self).__init__(config, key_name)
+
+    def exchange_code_for_login(self, app_config, http_client, code, redirect_suffix):
+
+        sub, lusername, email_address = super(RHSSOOAuthService, self).exchange_code_for_login(
+            app_config, http_client, code, redirect_suffix
```
Comment (kleesc) [inline oauth/services/rhsso.py:19]: Are these the correct references?

File: oauth/loginmanager.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
 from oauth.services.github import GithubOAuthService
 from oauth.services.google import GoogleOAuthService
+from oauth.services.rhsso import RHSSOOAuthService
 from oauth.oidc import OIDCLoginService
 
 CUSTOM_LOGIN_SERVICES = {
     "GITHUB_LOGIN_CONFIG": GithubOAuthService,
     "GOOGLE_LOGIN_CONFIG": GoogleOAuthService,
+    "RHSSO_LOGIN_CONFIG": RHSSOOAuthService,
```
Comment (kleesc) [inline oauth/loginmanager.py:9]: I don't think oidc implements the same interface as these services (`login_enabled`).
You might need to add an extra check when instantiating the oidc service.

**Review by kleesc — CHANGES_REQUESTED** [review]

File: oauth/services/rhsso.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import logging
+import os
+from _init import CONF_DIR
+
+logger = logging.getLogger(__name__)
+
+
+class RHSSOOAuthService(OIDCLoginService):
+    def exchange_code_for_login(self, app_config, http_client, code, redirect_suffix):
+
```
Comment (kleesc) [inline oauth/services/rhsso.py:14]: nit: extra empty line, otherwise LGTM

Comment (jonathankingfc) [reply oauth/services/rhsso.py:14]: This is from the code formatter

Comment (kleesc) [reply oauth/services/rhsso.py:14]: Ugh. ok...

**Review by kleesc — APPROVED** [review]

**Pull request merged**